transport/server.hh

```diff
@@ -148,6 +149,43 @@ struct connection_service_level_params {
     sstring scheduling_group_name;
 };
 
+class cql_metadata_id_wrapper {
+private:
+    std::optional<cql3::cql_metadata_id_type> _request_metadata_id;
+    std::optional<cql3::cql_metadata_id_type> _response_metadata_id;
+    bool _supported_by_protocol;
```

Comment:
The response_metadata_id doesn't need to be optional. If it wasn't requested, then we'll ignore it. Better to avoid wrapping it in yet another wrapper.

_supported_by_protocol is, I think also unnecessary. If it's not supported and the bit was set, the request should have been rejected. If it's supported then optional\<_request_metadata_id> conveys all the needed information.